### PR TITLE
Adds transfer token user fee

### PR DESCRIPTION
**Pull request by claravanstaden**
Adds user fee for transfer token on asset hub.

Snowbridge companion: https://github.com/Snowfork/snowbridge/pull/987
Resolves: [SNO-732](https://linear.app/snowfork/issue/SNO-732)

### CODE REVIEW

File: bridges/primitives/chain-asset-hub-rococo/src/lib.rs [hunk 1]
```diff
@@ -46,6 +46,10 @@ frame_support::parameter_types! {
 	/// Base delivery fee to `BridgeHubRococo`.
 	/// (initially was calculated by test `BridgeHubRococo::can_calculate_weight_for_paid_export_message_with_reserve_transfer`)
 	pub const BridgeHubRococoBaseFeeInRocs: u128 = 1214739988;
+
+	/// User fee for ERC20 token transfer back to Ethereum.
+	/// (initially was calculated by test `OutboundQueue::calculate_fees` - ETH/ROC 1/411 and fee_per_gas 15 GWEI = 18679250000 + *25%)
+	pub const TransferERC20TokenBaseFeeInRocs: u128 = 23349062500;
```
Comment (claravanstaden) [inline bridges/primitives/chain-asset-hub-rococo/src/lib.rs:52]: Not sure this is the best naming, I'm open to suggestions.

File: bridges/primitives/chain-asset-hub-rococo/src/lib.rs
Comment (vgeddes) [inline bridges/primitives/chain-asset-hub-rococo/src/lib.rs:1]: I'm not sure I would put our bridge stuff into crates in the `bridges` project. Its very specific to the polkadot-kusama bridge.

I would personally put this constant in a file like `snowbridge_config.rs` under `/cumulus/parachains/runtimes/assets/asset-hub/rococo/`

Then if we end up adding a whole more config, we can create a crate for it.

File: bridges/primitives/chain-asset-hub-rococo/src/lib.rs (hunk 1 above)
Comment (vgeddes) [reply bridges/primitives/chain-asset-hub-rococo/src/lib.rs:52]: I would suggest `BridgeHubEthereumBaseFeeInRocs` to be consistent with the Rocococ<->Wococo bridge.

File: bridges/primitives/chain-asset-hub-rococo/src/lib.rs
Comment (vgeddes) [reply bridges/primitives/chain-asset-hub-rococo/src/lib.rs:1]: We could also add other constants to the module, like  `EthereumGatewayAddress` to the module

File: bridges/primitives/chain-asset-hub-rococo/src/lib.rs [hunk 2]
```diff
@@ -46,6 +46,10 @@ frame_support::parameter_types! {
 	/// Base delivery fee to `BridgeHubRococo`.
 	/// (initially was calculated by test `BridgeHubRococo::can_calculate_weight_for_paid_export_message_with_reserve_transfer`)
 	pub const BridgeHubRococoBaseFeeInRocs: u128 = 1214739988;
+
+	/// User fee for ERC20 token transfer back to Ethereum.
+	/// (initially was calculated by test `OutboundQueue::calculate_fees` - ETH/ROC 1/411 and fee_per_gas 15 GWEI = 18679250000 + *25%)
```
Comment (vgeddes) [inline bridges/primitives/chain-asset-hub-rococo/src/lib.rs:51]: The fee parameters should result in a fee that is greater or equal to the fee generated by these parameters:

https://github.com/Snowfork/snowbridge/blob/2be6f57a2bc2c0e162cda32c331dafe5b80a8b80/parachain/pallets/outbound-queue/src/lib.rs#L252

So its a problem if fee_per_gas is 15gwei here and 30gwei on bridgehub.

Comment (claravanstaden) [reply bridges/primitives/chain-asset-hub-rococo/src/lib.rs:51]: I didn't realize those values in https://github.com/Snowfork/snowbridge/blob/2be6f57a2bc2c0e162cda32c331dafe5b80a8b80/parachain/pallets/outbound-queue/src/lib.rs#L252 are the final values I should use. I wrote a script that calculates the average gas price for the last three weeks like you suggested, and it was around 15 gwei. I'll change the calculation to use the values defined in the outbound queue then.

Comment (vgeddes) [reply bridges/primitives/chain-asset-hub-rococo/src/lib.rs:51]: Ah, rather update the values in the outbound queue, as your figure is more accurate :)

Comment (vgeddes) [reply bridges/primitives/chain-asset-hub-rococo/src/lib.rs:51]: To clarify, I wasn't saying those values were final, but rather that if we update the fees to be more accurate in one place, we must reconcile the fees in other places too.

Comment (claravanstaden) [reply bridges/primitives/chain-asset-hub-rococo/src/lib.rs:51]: Got it, agreed. Will add a comment about it too.

File: cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/src/xcm_config.rs [hunk 3]
```diff
@@ -226,9 +226,9 @@ impl Contains<RuntimeCall> for SafeCallFilter {
 					snowbridge_ethereum_beacon_client::Call::force_checkpoint { .. } |
 						snowbridge_ethereum_beacon_client::Call::set_operating_mode { .. },
 				) | RuntimeCall::EthereumInboundQueue(
-					snowbridge_inbound_queue::Call::set_operating_mode { .. },
+				snowbridge_inbound_queue::Call::set_operating_mode { .. },
```
Comment (claravanstaden) [inline cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/src/xcm_config.rs:229]: fmt

**Review by vgeddes — APPROVED** [review]

**Pull request merged**